### PR TITLE
build: add release-tools files and makefile reference

### PR DESCRIPTION
**Pull request by hairyhum**
Integrating with https://github.com/kubernetes-csi/csi-release-tools tooling.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,8 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Protect against invocation without a specific target.
-all:
+all: build
```
Comment (carlbraganza) [inline Makefile:15]: Is `all` a standard target for the CI?

**Review by carlbraganza — CHANGES_REQUESTED** [review]
Doesn't look like `all` is a standard target so don't remove the protection in the top level `Makefile`.

File: Makefile (hunk 1 above)
Comment (hairyhum) [reply Makefile:15]: My understanding is that `prow` is requiring that https://github.com/kubernetes-csi/external-snapshotter/blob/master/release-tools/prow.sh#L1247

Comment (carlbraganza) [reply Makefile:15]: Hmm... that indeed appears to be the case. So I withdraw my objection!

**Review by carlbraganza — COMMENTED** [review]
/lgtm

**Review by carlbraganza — APPROVED** [review]
I withdraw my comment on the `all` target.

**Review by carlbraganza — APPROVED** [review]
/lgtm

File: release-tools/go-modules-update.sh [hunk 2]
```diff
@@ -0,0 +1,129 @@
+#!/bin/sh
+
+# Copyright 2023 The Kubernetes Authors.
```
Comment (PrasadG193) [inline release-tools/go-modules-update.sh:3]:
```suggestion
# Copyright 2024 The Kubernetes Authors.
```

**Review by PrasadG193 — COMMENTED** [review]
Lots of license headers has incorrect year, we might want to correct that

**Pull request merged**